### PR TITLE
Setup for Weekly Debrief

### PR DESCRIPTION
**Pull request by dishanest**
## Overview
Setup models and networking for Weekly Debrief

## Changes Made
- Add `GetWeeklyDebrief` query
- Add `WeeklyDebrief` model struct
- Update `incrementShoutouts` call to include uuid

## Test Coverage
Built to ensure changes didn't break anything

## Next Steps
- WD entry point on home screen
- @amyqxhuang WD popup

### CODE REVIEW

File: Volume/Supporting/UserData.swift [hunk 1]
```diff
@@ -120,7 +129,7 @@ class UserData: ObservableObject {
     }
 
     func set(publication: Publication, isFollowed: Bool) {
-        guard let uuid = UserDefaults.standard.string(forKey: userUUIDKey) else {
+        guard let uuid = self.uuid else {
```
Comment (danielVebman) [inline Volume/Supporting/UserData.swift:132]: Delete self?

File: Volume/Supporting/UserData.swift [hunk 2]
```diff
@@ -50,6 +50,15 @@ class UserData: ObservableObject {
         }
     }
     
+    private(set) var uuid: String? {
+        get {
+            return UserDefaults.standard.object(forKey: userUUIDKey) as? String
```
Comment (danielVebman) [inline Volume/Supporting/UserData.swift:55]:
```suggestion
           UserDefaults.standard.object(forKey: userUUIDKey) as? String
```
Implicit return

File: Volume/Models/WeeklyDebrief.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    var numBookmarkedArticles: Int
+    var readArticles: [Article]
+    var randomArticles: [Article]
+    
+    init(from weeklyDebrief: WeeklyDebriefFields) {
+        /*
+         schema changes required:
+         - createdAt -> creationDate
+         - numShoutouts, numReadArticles, numBookmarkedArticles -> type Int
+         */
```
Comment (danielVebman) [inline Volume/Models/WeeklyDebrief.swift:26]: Why is this here? You shouldn't commit a partial feature to main. If it is a work in progress, then make it a draft and don't merge until everything is done.

File: Volume/Models/WeeklyDebrief.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import Foundation
+
+struct WeeklyDebrief {
+    var creationDate: Date
+    var expirationDate: Date
+    var numShoutouts: Int
+    var numReadArticles: Int
+    var numBookmarkedArticles: Int
+    var readArticles: [Article]
+    var randomArticles: [Article]
```
Comment (danielVebman) [inline Volume/Models/WeeklyDebrief.swift:19]: Can these not be `let`?

**Review by danielVebman — CHANGES_REQUESTED** [review]
Is this feature incomplete? See my comment

**Review by amyqxhuang — APPROVED** [review]
looks good to me!
